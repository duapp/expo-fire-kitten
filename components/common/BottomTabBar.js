import React from 'react';
import { BottomNavigation, BottomNavigationTab } from '@ui-kitten/components';
import { SafeAreaView } from 'react-native-safe-area-context';

export default function BottomTabBar({ navigation, state }) {
  return (
    <SafeAreaView edges={['bottom']}>
      <BottomNavigation
        selectedIndex={state.index}
        onSelect={index => navigation.navigate(state.routeNames[index])}
      >
        <BottomNavigationTab title="Home" />
        <BottomNavigationTab title="Test" />
      </BottomNavigation>
    </SafeAreaView>
  );
}
