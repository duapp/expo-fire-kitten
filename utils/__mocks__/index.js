import React from 'react';
import { Text } from 'react-native';

/**
 * Mock icon helper to use text instead of icon for testing purpose
 * @param {string} name Icon name
 * @returns Text element with the icon name
 */
export function EvaIcon(name) {
  return () => <Text>{name}</Text>;
}

/**
 * Mock translation helper to simply return the key for testing purpose
 * @param {string | string[]} scope
 * @returns Simple translation key
 */
export function t(scope) {
  return scope;
}

/**
 * Mock helper to return the mock data for the db path
 * @param {string} path in Realtime DB
 * @returns {object} value
 */
export function read(path) {
  return new Promise(resolve => {
    resolve({
      value: path,
    });
  });
}
