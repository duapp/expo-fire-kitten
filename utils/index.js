import React from 'react';
import { Icon } from '@ui-kitten/components';
import i18n from '../config/i18n';
import firebase from '../config/firebase';

/**
 * Creates an Icon component based on the Eva icon name
 * @param {string} name Icon name from https://akveo.github.io/eva-icons/
 * @returns Icon component
 */
export function EvaIcon(name) {
  /* eslint-disable-next-line react/jsx-props-no-spreading */
  return props => <Icon {...props} name={name} />;
}

/**
 * Alias for i18n.t: get translation string
 * @param {string | string[]} scope translation key
 * @param {object} options translation options
 * @returns {string} translated string
 */
export function t(scope, options) {
  return i18n.t(scope, options);
}

/**
 * Read data from Firebase Realtime DB by the path
 * @param {string} path
 * @returns {object} object value at the path
 */
export async function read(path) {
  const ref = firebase.database().ref();
  const snapshot = await ref.child(path).get();
  return snapshot.val();
}
