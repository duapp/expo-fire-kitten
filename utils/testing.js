import React from 'react';
import * as eva from '@eva-design/eva';
import { ApplicationProvider } from '@ui-kitten/components';

export function TestRoot(props) {
  return (
    <ApplicationProvider {...eva} theme={eva.light}>
      {props.children}
    </ApplicationProvider>
  );
}
