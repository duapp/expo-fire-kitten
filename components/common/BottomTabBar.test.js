import React from 'react';
import { render } from '@testing-library/react-native';
import BottomTabBar from './BottomTabBar';
import { TestRoot } from '../../utils/testing';

const mockState = {
  index: 0,
  routeNames: ['Home']
};

test('it works', () => {
  const { getByText, toJSON } = render(
    <TestRoot>
      <BottomTabBar state={mockState} />
    </TestRoot>
  );

  expect(getByText('Home')).toBeEnabled();
  expect(toJSON()).toMatchSnapshot();
});
