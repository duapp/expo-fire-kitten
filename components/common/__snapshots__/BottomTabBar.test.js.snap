// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it works 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <RNCSafeAreaView>
    <View
      appearance="default"
      onSelect={[Function]}
      selectedIndex={0}
      style={
        Array [
          Object {
            "backgroundColor": "#FFFFFF",
            "minHeight": 56,
            "paddingVertical": 4,
          },
          Object {
            "flexDirection": "row",
          },
          undefined,
        ]
      }
    >
      <View
        onLayout={[Function]}
        positions={2}
        selectedPosition={0}
        style={
          Object {
            "backgroundColor": "#3366FF",
            "height": 4,
            "position": "absolute",
            "transform": Array [
              Object {
                "translateX": 0,
              },
            ],
            "width": "50%",
          }
        }
      />
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "flex": 1,
            "justifyContent": "center",
            "opacity": 1,
          }
        }
      >
        <Text
          appearance="default"
          category="p1"
          style={
            Array [
              Object {
                "color": "#222B45",
                "fontFamily": "System",
                "fontSize": 15,
                "fontWeight": "400",
              },
              Object {
                "color": "#3366FF",
                "fontFamily": "System",
                "fontSize": 12,
                "fontWeight": "600",
                "marginVertical": 2,
              },
            ]
          }
        >
          Home
        </Text>
      </View>
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "flex": 1,
            "justifyContent": "center",
            "opacity": 1,
          }
        }
      >
        <Text
          appearance="default"
          category="p1"
          style={
            Array [
              Object {
                "color": "#222B45",
                "fontFamily": "System",
                "fontSize": 15,
                "fontWeight": "400",
              },
              Object {
                "color": "#8F9BB3",
                "fontFamily": "System",
                "fontSize": 12,
                "fontWeight": "600",
                "marginVertical": 2,
              },
            ]
          }
        >
          Test
        </Text>
      </View>
    </View>
  </RNCSafeAreaView>
</View>
`;
